src/Plugins/Loader/ManagedLoadContext.cs

```diff
@@ -309,7 +317,62 @@ private bool SearchForLibrary(NativeLibrary library, string prefix, out string?
         private IntPtr LoadUnmanagedDllFromResolvedPath(string unmanagedDllPath)
         {
             var normalized = Path.GetFullPath(unmanagedDllPath);
-            return LoadUnmanagedDllFromPath(normalized);
+
+            // Create shadow copy directory only when hot reloading is enabled and plugins are
+            // dependent on unmanaged DLL(s).
+            if (_loadInMemory && string.IsNullOrEmpty(_unmanagedDllShadowCopyDirectoryPath))
```

Comment:
There is a possible race condition here if multiple threads are loading different unmanaged dlls at the same time. I think this would be addressed by initializing _unmanagedDllShadowCopyDirectoryPath in the constructor.